New Mode Team-Tournament

eg3r opened the pull request:
This PR is about adding a new mode "**TeamTournament**".

- A lot of changes were done, basically a complete new tournament mode was implemented, codebase has tripled :see_no_evil:.
- Adds a new setting to enabe/disable the option for "joining as team"
- Randomized start with up to 32 Teams each up to 8 members (randomized per tournament)
- Uses the Bandit-Camp selection screen for team selection, but with his own VM
- Does not change the basic tournament mode 
- Adds Patches for two functions but only executed if a TeamTournament is selected
- No savegame changes
- Issues still to address : 
  - Does not apply joining of heroes based on tournament type (Lord...) this is a trivial matter and will/can be done shortly
  - Win View is the old one, in general all views are old just with new VMs
  - Right now values for TeamSize, Initial groups etc. are randomized and maybe need adjusting
- I probably will test more and apply fixes on the go.
- As a complete tournament mode is now implemeneted, all changes are possible without patching (mostly)

PS. also fixes #7

eg3r committed:
Fixed perk Duelist -> giving double renown for win

eg3r committed:
Direct edits in github :/

eg3r committed:
Initial TeamTournament Mode Commit

This adds the basic team tournament mode

eg3r committed:
Changed bet odd calc for now, still needs rework

alexhuijsman commented:
Awesome work! However, I am currently getting a build error: 
`TournamentsEnhanced/src/TeamTournament/Menu/ViewModels/ManageTeamSelectionVM.cs(44,28): error CS1061: 'Hero' does not contain a definition for 'AllLivingRelatedHeroes' and no accessible extension method 'AllLivingRelatedHeroes' accepting a first argument of type 'Hero' could be found (are you missing a using directive or an assembly reference?) [/home/alex/code/TournamentsEnhanced/TournamentsEnhanced.csproj]`

eg3r committed:
Made only the team appear in the view, added missing function

eg3r committed:
Merge branch '1.5.8' of https://github.com/3ger/TournamentsEnhanced into 1.5.8

eg3r commented:
For some reason my diff missed a file, added now. 
Also I fixed the cluttering in the view when huge teams were made, by just having one Hero appear as the Team (represented by the Team-Leader).
